### ISSUE TITLE
[sklearn#33712] Add replaced_undefined_by in classification metrics

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#382]
**Mirrored from [scikit-learn#33712](https://github.com/scikit-learn/scikit-learn/issues/33712)**

Original title: Add replaced_undefined_by in classification metrics
Original labels: Enhancement, API, Meta-issue
Created: 2026-04-08T21:52:06Z

---

Following https://github.com/scikit-learn/scikit-learn/issues/29048 and in particular https://github.com/scikit-learn/scikit-learn/issues/29048#issuecomment-2857931743, we want to replace ``zero_division`` by a ``replaced_undefined_by`` parameter to the following classification metrics:

- [ ] ``jaccard_score``
- [ ] ``f1_score``
- [ ] ``fbeta_score``
- [ ] ``precision_recall_fscore_support``
- [ ] ``precision_score``
- [ ] ``recall_score``
- [ ] ``classification_report``

In these metrics, ``zero_division`` should be deprecated and ``replaced_undefined_by`` with the following behavior:
- default to ``nan``
- when the metric is undefined: raise an UndefinedMetricWarning and return ``replaced_undefined_by``
- the warning message should explain the reason for the metric to be undefined

We also want to add a ``replaced_undefined_by`` parameter the following classification metrics:
- [ ] ``balanced_accuracy_score``
- [ ] ``matthews_corrcoef``

We may also want to add the param to some metrics in ``_ranking.py``:
- [ ] ``ndcg_score``

(The lists will be updated when we identify metrics that need it)

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to [12](https://github.com/godofecht/flow-scikit/issues/12) in the flow-scikit repository. Further library-specific triage and implementation will happen there.

**Comment by godofecht** [comment]
Moved to [14](https://github.com/godofecht/flow-scikit/issues/14) in the flow-scikit repository. Further library-specific triage and implementation will happen there.
